### PR TITLE
docs: rework README and contributing docs

### PR DESCRIPTION
**Pull request by neersighted**
Rework the README, contributing docs, and issue templates based on the current usage and direction of the project.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/---documentation.md [hunk 1]
```diff
@@ -1,22 +1,21 @@
...
+    Before you submit your issue, please make sure you have completed the checklist below.
 -->
 
-<!-- Checked checkbox should look like this: [x] -->
+<!-- All the below steps should be completed before submitting your issue. Checked checkbox should look like this: [x] -->
 - [ ] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
 
 ## Issue
-<!-- Now feel free to write your issue, but please be descriptive! Thanks again 🙌 ❤️ -->
+<!-- Now feel free to write your issue, and please be descriptive as possible! -->
```
Comment (mkniewallner) [inline .github/ISSUE_TEMPLATE/---documentation.md:20]:
```suggestion
<!-- Now feel free to write your issue, and please be as descriptive as possible! -->
```

File: .github/ISSUE_TEMPLATE/---feature-request.md [hunk 2]
```diff
@@ -1,23 +1,23 @@
...
+about: Do you have ideas for new features or improvements?
 title: ''
 labels: kind/feature, status/triage
 assignees: ''
 
 ---
 
 <!--
-  Hi there! Thank you for wanting to make Poetry better.
+    Hi there! Thank you for submitting a bug report!
```
Comment (mkniewallner) [inline .github/ISSUE_TEMPLATE/---feature-request.md:11]:
```suggestion
    Hi there! Thank you for submitting a feature request!
```

File: .github/ISSUE_TEMPLATE/---feature-request.md [hunk 3]
```diff
@@ -1,23 +1,23 @@
...
 
-  Before you submit this; let's make sure of a few things.
-  Please make sure the following boxes are ticked if they are correct.
-  If not, please try and fulfill these first.
+    Before you submit your issue, please review and follow the instructions at
+    https://python-poetry.org/docs/contributing/#suggesting-enhancements.
 -->
 
-<!-- Checked checkbox should look like this: [x] -->
+<!-- All the below steps should be completed before submitting your issue. Checked checkbox should look like this: [x] -->
```
Comment (mkniewallner) [inline .github/ISSUE_TEMPLATE/---feature-request.md:17]:
```suggestion
<!-- All the below steps should be completed before submitting your request. Checked checkbox should look like this: [x] -->
```

File: .github/ISSUE_TEMPLATE/---feature-request.md [hunk 4]
```diff
@@ -1,23 +1,23 @@
...
 - [ ] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
-- [ ] I have searched the [documentation](https://python-poetry.org/docs/) and believe that my question is not covered.
+- [ ] I have searched the [FAQ](https://python-poetry.org/docs/faq/) and general [documentation](https://python-poetry.org/docs/) and believe that my question is not already covered.
```
Comment (mkniewallner) [inline .github/ISSUE_TEMPLATE/---feature-request.md:19]:
```suggestion
- [ ] I have searched the [FAQ](https://python-poetry.org/docs/faq/) and general [documentation](https://python-poetry.org/docs/) and believe that my request is not already covered.
```

File: .github/ISSUE_TEMPLATE/---feature-request.md [hunk 5]
```diff
@@ -1,23 +1,23 @@
...
 
 ## Feature Request
-<!-- Now feel free to write your idea for improvement. Thanks again 🙌 ❤️ -->
+<!-- Now feel free to write your issue, and please be descriptive as possible! -->
```
Comment (mkniewallner) [inline .github/ISSUE_TEMPLATE/---feature-request.md:22]:
```suggestion
<!-- Now feel free to write your feature request, and please be as descriptive as possible! -->
```

File: README.md [hunk 6]
```diff
@@ -1,34 +1,18 @@
...
+[![Pre-release Version](https://img.shields.io/github/v/release/python-poetry/poetry?label=pre-release&include_prereleases&sort=semver)][PyPI Releases]
+[![Python Versions](https://img.shields.io/pypi/pyversions/poetry)][PyPI]
+[![Download Stats](https://img.shields.io/pypi/dm/poetry)](https://pypistats.org/packages/poetry)
+[![Discord](https://img.shields.io/discord/487711540787675139?logo=discord)][Discord]
 
 Poetry helps you declare, manage and install dependencies of Python projects,
 ensuring you have the right stack everywhere.
 
-It requires Python 3.7+ to run.
+![Poetry Install](assets/install.gif)
```
Comment (mkniewallner) [inline README.md:12]: Won't doing that break the gif when displayed [in PyPI](https://pypi.org/project/poetry/)?

File: README.md [hunk 7]
```diff
@@ -42,68 +26,89 @@ authors = [
...
-
-`poetry` will also detect if you are inside a virtualenv and install the packages accordingly.
-So, `poetry` can be installed globally and used everywhere.
-
-`poetry` also comes with a full-fledged dependency resolution library.
+## Installation
 
-## Why?
+Poetry supports multiple installation methods, including a simple script found at [install.python-poetry.org]. For full
+installation instructions, including advanced usage of the script, alternate install methods, and CI best practices, see
```
Comment (mkniewallner) [inline README.md:71]:
```suggestion
installation instructions, including advanced usages of the script, alternate install methods, and CI best practices, see
```

File: docs/basic-usage.md [hunk 8]
```diff
@@ -55,10 +55,16 @@ requires = ["poetry-core"]
 build-backend = "poetry.core.masonry.api"
 ```
 
-Poetry assumes your package contains a package with the same name as `tool.poetry.name`.
-If this is not the case, populate `tool.poetry.packages` to specify your package or packages.
+Poetry assumes your package contains a package with the same name as `tool.poetry.name` located in the root of your
+project.  If this is not the case, populate [`tool.poetry.packages`]({{< relref "pyproject#packages" >}}) to specify
```
Comment (mkniewallner) [inline docs/basic-usage.md:59]:
```suggestion
project. If this is not the case, populate [`tool.poetry.packages`]({{< relref "pyproject#packages" >}}) to specify
```

File: docs/basic-usage.md [hunk 9]
```diff
@@ -92,17 +98,30 @@ $ poetry add pendulum
 
 It will automatically find a suitable version constraint **and install** the package and sub-dependencies.
 
+Poetry supports a rich [dependency specification]({{< relref "dependency-specification" >}}) syntax, including caret,
+tilde, wildcard, inequality and
+[multiple constraint]({{< relref "dependency-specification#multiple-constraints-dependencies" >}}) requirements.
```
Comment (mkniewallner) [inline docs/basic-usage.md:103]:
```suggestion
[multiple constraints]({{< relref "dependency-specification#multiple-constraints-dependencies" >}}) requirements.
```

File: docs/basic-usage.md [hunk 10]
```diff
@@ -92,17 +98,30 @@ $ poetry add pendulum
...
 
 ## Using your virtual environment
 
-By default, poetry creates a virtual environment in `{cache-dir}/virtualenvs` (`{cache-dir}\virtualenvs` on Windows).
-You can change the [`cache-dir`]({{< relref "configuration#cache-dir" >}} "cache-dir configuration documentation") value by editing the poetry config.
-Additionally, you can use the [`virtualenvs.in-project`]({{< relref "configuration#virtualenvsin-project" >}} "#virtualenvs.in-project configuration documentation") configuration variable
-to create virtual environment within your project directory.
-
+By default, poetry creates a virtual environment in `{cache-dir}/virtualenvs`.
```
Comment (mkniewallner) [inline docs/basic-usage.md:107]:
```suggestion
By default, Poetry creates a virtual environment in `{cache-dir}/virtualenvs`.
```

File: docs/basic-usage.md [hunk 11]
```diff
@@ -92,17 +98,30 @@ $ poetry add pendulum
...
+You can change the [`cache-dir`]({{< relref "configuration#cache-dir" >}} "cache-dir configuration documentation") value
+by editing the Poetry config.
```
Comment (mkniewallner) [inline docs/basic-usage.md:109]:
```suggestion
by editing the Poetry configuration.
```

File: docs/basic-usage.md [hunk 12]
```diff
@@ -92,17 +98,30 @@ $ poetry add pendulum
...
+Additionally, you can use the
+[`virtualenvs.in-project`]({{< relref "configuration#virtualenvsin-project" >}}) configuration variable to create
+virtual environment within your project directory.
```
Comment (mkniewallner) [inline docs/basic-usage.md:112]:
```suggestion
virtual environments within your project directory.
```

File: docs/contributing.md [hunk 13]
```diff
@@ -7,275 +7,256 @@ layout: single
 menu:
   docs:
     weight: 100
----
 
-<!--
-The content of this file is manually kept in sync with ../CONTRIBUTING.md. There are formatting differences because this file uses Hugo templating.
--->
+note: "Are you viewing this document on Github? For the best experience, view it on the website https://python-poetry.org/docs/contributing."
```
Comment (mkniewallner) [inline docs/contributing.md:11]:
```suggestion
note: "Are you viewing this document on GitHub? For the best experience, view it on the website https://python-poetry.org/docs/contributing."
```

File: docs/contributing.md [hunk 14]
```diff
@@ -7,275 +7,256 @@ layout: single
...
+* **If the problem involves an unexpected error being raised**, execute the problematic command in **debug** mode
+(with `-vvv` flag).
+
+Provide detailed steps for reproduction of your issue:
+
+* **Provide your pyproject.toml file** in a [Gist](https://gist.github.com), pastebin, or example repository after
+removing potential private information like private package repositories or names.
+* **Provide specific examples to demonstrate the steps to reproduce the issue**. This could be an example repository, a
+sequence of steps run in a container, or just a pyproject.toml for very simple cases.
+* **Are you unable reliably reproduce the issue?** If so, provide details about how often the problem happens
```
Comment (mkniewallner) [inline docs/contributing.md:62]:
```suggestion
* **Are you unable to reliably reproduce the issue?** If so, provide details about how often the problem happens
```

File: docs/contributing.md [hunk 15]
```diff
@@ -7,275 +7,256 @@ layout: single
...
+inside that container/VM in your issue report.
 
 ### Suggesting enhancements
 
-This section guides you through submitting an enhancement suggestion for Poetry, including completely new features and minor improvements to existing functionality. Following these guidelines helps maintainers and the community understand your suggestion and find related suggestions.
-
-Before creating enhancement suggestions, please check [this list](#before-submitting-an-enhancement-suggestion) as you might find out that you don't need to create one. When you are creating an enhancement suggestion, please [include as many details as possible](#how-do-i-submit-an-enhancement-suggestion). Fill in [the template](https://github.com/python-poetry/poetry/blob/master/.github/ISSUE_TEMPLATE/---feature-request.md), including the steps that you imagine you would take if the feature you're requesting existed.
+This section guides you through submitting an enhancement suggestion for Poetry, including completely new features as
+well as improvements to existing functionality. Following these guidelines helps maintainers and the community
```
Comment (mkniewallner) [inline docs/contributing.md:88]:
```suggestion
well as improvements to existing functionalities. Following these guidelines helps maintainers and the community
```

File: docs/contributing.md [hunk 16]
```diff
@@ -7,275 +7,256 @@ layout: single
...
+understand your suggestion and find related suggestions.
 
-#### Before submitting an enhancement suggestion
+#### Before submitting an suggested enhancement
```
Comment (mkniewallner) [inline docs/contributing.md:91]:
```suggestion
#### Before submitting a suggested enhancement
```

File: docs/contributing.md [hunk 17]
```diff
@@ -7,275 +7,256 @@ layout: single
...
 
-* **Check the [FAQs on the official website](https://python-poetry.org/docs/faq)** for a list of common questions and problems.
-* **Check that your issue does not already exist in the [issue tracker](https://github.com/python-poetry/poetry/issues)**.
+* **Check the [FAQ]** for a list of common questions and problems.
+* **Check that your issue does not already exist** in the [issue tracker].
 
-#### How do I submit an Enhancement suggestion?
+#### How do I submit an suggested enhancement?
```
Comment (mkniewallner) [inline docs/contributing.md:96]:
```suggestion
#### How do I submit a suggested enhancement?
```

File: docs/contributing.md [hunk 18]
```diff
@@ -7,275 +7,256 @@ layout: single
...
+start a [Discussion][Discussions].
 
 #### Local development
 
-You will need Poetry to start contributing on the Poetry codebase. Refer to the [documentation](https://python-poetry.org/docs/#introduction) to start using Poetry.
+Poetry is developed using Poetry. Refer to the [documentation] to install Poetry in your local environment.
 
 {{% note %}}
-Local development of Poetry requires Python 3.8 or newer.
+Poetry's development toolchain requires Python 3.8 or newer.
```
Comment (mkniewallner) [inline docs/contributing.md:141]:
```suggestion
Poetry's development toolchain requires Python 3.7 or newer.
```

File: docs/contributing.md [hunk 19]
```diff
@@ -7,275 +7,256 @@ layout: single
...
 {{% /note %}}
 
-You will first need to clone the repository using `git` and place yourself in its directory:
+You should first fork the Poetry repository and then clone it locally, so that you can make pull requests against the
+project. If you are new to Git and pull request based development, Github provides a
```
Comment (mkniewallner) [inline docs/contributing.md:145]:
```suggestion
project. If you are new to Git and pull request based development, GitHub provides a
```

File: docs/contributing.md [hunk 20]
```diff
@@ -7,275 +7,256 @@ layout: single
...
 #### Pull requests
 
-* Fill in [the required template](https://github.com/python-poetry/poetry/blob/master/.github/PULL_REQUEST_TEMPLATE.md)
-* Be sure that your pull request contains tests that cover the changed or added code.
-* If your changes warrant a documentation change, the pull request must also update the documentation.
+* Fill out the pull request body completely and describe your changes as accurately as possible. The pull request body
+should be kept up to date as it will usually form the base for the final merge commit and the changelog entry.
+* Be sure that your pull request contains tests that cover the changed or added code. Tests are generally required for
+code be to be considered mergable, and code without passing tests will not be merged.
+* Ensure your pull requests passes the mypy and pre-commit checks pass. Remember that you can run these tools locally
```
Comment (mkniewallner) [inline docs/contributing.md:183]:
```suggestion
* Ensure your pull request passes the mypy and pre-commit checks pass. Remember that you can run these tools locally
```

File: docs/contributing.md [hunk 21]
```diff
@@ -7,275 +7,256 @@ layout: single
...
-All pull requests, unless otherwise instructed, need to be first accepted into the main branch (`master`).
+All pull requests, unless otherwise instructed, need to be first accepted into the `master` branch. Maintainers will
+generally decide if any backports to other branches are required, and carry them out as needed.
 
 ### Issue triage
 
 {{% note %}}
-If you have an issue that hasn't had any attention, you can ping us `@python-poetry/triage` on the issue. Please, give us reasonable time to get to your issue first, spamming us with messages
+If you have an issue that hasn't had any attention, you can ping us `@python-poetry/triage` on the issue. Please give us
+reasonable time to get to your issue first, and avoid pinging any individuals directly, especially if they are not part
```
Comment (mkniewallner) [inline docs/contributing.md:201]:
```suggestion
reasonable time to get to your issue first, and avoid pinging any individual directly, especially if they are not part
```

File: docs/contributing.md [hunk 22]
```diff
@@ -7,275 +7,256 @@ layout: single
...
-   1. The pull request must have the `Release` label specified.
-
-Once the branch pull-request is ready and approved, a member of `@python-poetry/core` will,
-
-1. Tag the branch with the version identifier (eg: `1.1.0rc1`).
-2. Merge the pull request once the release is created and assets are uploaded by the CI.
-
 {{% note %}}
-In this case, we prefer a merge commit instead of squash or rebase merge.
+This mechanism can also be used to test pull requests by using Github's pull request remote refs:
```
Comment (mkniewallner) [inline docs/contributing.md:248]:
```suggestion
This mechanism can also be used to test pull requests by using GitHub's pull request remote refs:
```

File: docs/contributing.md (hunk 21 above)
Comment (neersighted) [reply docs/contributing.md:201]: this is correct

File: .github/ISSUE_TEMPLATE/---feature-request.md (hunk 3 above)
Comment (neersighted) [reply .github/ISSUE_TEMPLATE/---feature-request.md:17]: I like issue as a generic descriptor of anything on the issue tracker

File: .github/ISSUE_TEMPLATE/---feature-request.md (hunk 4 above)
Comment (neersighted) [reply .github/ISSUE_TEMPLATE/---feature-request.md:19]: I think this works fine

File: README.md (hunk 6 above)
Comment (neersighted) [reply README.md:12]: Ah, shoot, good point... Might just need to revert this.

File: README.md (hunk 7 above)
Comment (neersighted) [reply README.md:71]: singular usage is more correct here

File: docs/contributing.md (hunk 15 above)
Comment (neersighted) [reply docs/contributing.md:88]: singular is more correct

File: docs/contributing.md (hunk 21 above)
Comment (mkniewallner) [reply docs/contributing.md:201]: Oh ok, sorry, it sounded weird as a non-native English speaker. [TIL](https://english.stackexchange.com/questions/111444/any-individual-or-any-individuals) :slightly_smiling_face:

File: docs/contributing.md (hunk 18 above)
Comment (mkniewallner) [reply docs/contributing.md:141]: Ah, blaming led me to https://github.com/snok/flake8-type-checking, so now I see why it's 3.8+. My bad.

**Review by mkniewallner — APPROVED** [review]
:tada:

**Pull request merged**